corgi/tasks/errata_tool.py

```
@@ -52,45 +52,60 @@ def slow_load_errata(erratum_name, force_process: bool = False):
-    relation_build_ids = _get_relation_build_ids(erratum_id)
+    relation_builds = _get_relation_builds(erratum_id)
+
+    build_types = set()
+    build_ids = set()
+    for build_id, build_type in relation_builds:
+        build_ids.add(build_id)
+        build_types.add(build_type)
+
+    # Errata are not used in community products, and we are not yet attaching other builds types
+    # like HACBS to errata. If that changes, we should review this task for correctness
+    if len(build_types) > 1:
+        raise ValueError("Multiple build types found for errata %s", erratum_name)
 
     # Check is we have software builds for all of them.
     # Most of the time will not have all the builds
     # so we don't load all the software builds at this point
-    no_of_processed_builds = SoftwareBuild.objects.filter(build_id__in=relation_build_ids).count()
+    no_of_processed_builds = SoftwareBuild.objects.filter(
+        build_id__in=build_ids, build_type=BUILD_TYPE
+    ).count()
 
     # If we have no relations at all, or we want to update them
-    if len(relation_build_ids) == 0 or force_process:
+    if len(build_ids) == 0 or force_process:
         # Save PCR
         logger.info("Saving product components for errata %s", erratum_id)
         variant_to_component_map = et.get_erratum_components(str(erratum_id))
         for variant_id, build_objects in variant_to_component_map.items():
             for build_obj in build_objects:
                 for build_id, errata_components in build_obj.items():
                     # Add to relations list as we go, so we can fetch them below
-                    relation_build_ids.add(int(build_id))
+                    build_ids.add(int(build_id))
                     ProductComponentRelation.objects.update_or_create(
                         external_system_id=erratum_id,
                         product_ref=variant_id,
                         build_id=build_id,
+                        build_type=BUILD_TYPE,
```

As above - let's make sure we use the right build_type. Or maybe we only expect to call this code for RH Brew, since the logic here is for Errata Tool? In that case, let's raise an error if the one build_type we have isn't equal to SoftwareBuild.Type.BREW.

---

We only expect to process Errata in Enterprise mode, where `BUILD_TYPE == "BREW"`. Now raising an error if `BUILD_TYPE != "BREW"`